tests/integration/routes/test_questionnaire.py

```diff
@@ -53,3 +55,39 @@ def test_options_request_before_request(self):
 
         for output in logs.output:
             self.assertNotIn("questionnaire request", output)
+
+    def test_no_state_before_request(self):
+        # Given
+        self.launchSurvey("test_view_submitted_response")
+
+        # When
+        with patch("app.routes.questionnaire.get_metadata", return_value=None):
+            self.post()
+
+            # Then
+            self.assertStatusUnauthorised()
+
+    def test_questionnaire_not_submitted_no_state_before_submission_request(self):
+        # Given
+        self.launchSurvey("test_view_submitted_response")
+
+        # When
+        with patch("app.routes.questionnaire.get_metadata", return_value=None):
+            self.get("/submitted/view-response")
+
+            # Then
+            self.assertStatusUnauthorised()
+
+    def test_questionnaire_submitted_no_state_before_submission_request(self):
+        # Given
+        self.launchSurvey("test_view_submitted_response")
+        self.post()
+        self.post()
+
+        # When
+        with patch("app.routes.questionnaire.get_metadata", return_value=None):
+            self.get("/submitted/view-response")
+
+            # Then
+            self.assertStatusUnauthorised()
+            self.assertInBody("Your questionnaire has been submitted")
```

Comment:
Some of these tests shouldn't be in this file as they are not all testing `questionnaire` routes. I do think they should be kept together though, so maybe they could be moved to a new test file, something like `test_no_questionnaire_state.py`.